### PR TITLE
Disable progress bar for small fetchings

### PR DESCRIPTION
**Pull request by AleksanderWWW**
## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [ ] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

File: src/neptune/internal/backends/hosted_neptune_backend.py [hunk 1]
```diff
@@ -1094,7 +1094,7 @@ def search_leaderboard_entries(
                 sort_by=sort_by,
                 ascending=ascending,
                 sort_by_column_type=sort_by_column_type,
-                progress_bar=progress_bar,
+                progress_bar=False if limit < default_step_size else progress_bar,
```
Comment (AleksanderWWW) [inline src/neptune/internal/backends/hosted_neptune_backend.py:1097]: If `limit` is smaller than the `default_step_size` then it means that we will not fetch even a "full" single page, so no need to construct a progress bar for that

Comment (Raalsky) [reply src/neptune/internal/backends/hosted_neptune_backend.py:1097]: But as we're always fetching the total number of matching entries -> is it needed?

File: src/neptune/api/searching_entries.py [hunk 2]
```diff
@@ -154,19 +154,19 @@ def iter_over_pages(
...
 
-    total = 0
+    total = get_single_page(
+        limit=0,
+        offset=0,
+        **kwargs,
+    ).get("matchingItemCount", 0)
 
-    progress_bar = progress_bar if step_size >= total else None
+    progress_bar = False if total < step_size else progress_bar  # disable progress bar if only one page is fetched
```
Comment (Raalsky) [inline src/neptune/api/searching_entries.py:163]: I'm not 100% sure but shouldn't it be `<=`?

File: src/neptune/internal/backends/hosted_file_operations.py [hunk 3]
```diff
@@ -481,6 +481,9 @@ def _store_response_as_file(
         target_file = destination
 
     total_size = int(response.headers.get("content-length", 0))
+
+    progress_bar = False if total_size < 1024 * 1024 else progress_bar  # less than one chunk
```
Comment (Raalsky) [inline src/neptune/internal/backends/hosted_file_operations.py:485]: 1. Magic numbers 🛑 
2. Looks like`content-length` may not be provided. If I understand correctly this doesn't mean that the file is empty; rather it's not possible to determine the target size. In such a case we should probably show the progress bar.

File: src/neptune/api/searching_entries.py (hunk 2 above)
Comment (AleksanderWWW) [reply src/neptune/api/searching_entries.py]: Done

File: src/neptune/internal/backends/hosted_file_operations.py (hunk 3 above)
Comment (AleksanderWWW) [reply src/neptune/internal/backends/hosted_file_operations.py]: Done

**Review by Raalsky — APPROVED** [review]

**Pull request merged**